### PR TITLE
feat: Component design system

### PR DESCRIPTION
**Pull request by stepinfwd**
Creating storybook , integration of shadecn , and development for reusable button component

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -9,8 +9,10 @@
     "@emoji-mart/data": "^1.1.2",
     "@emoji-mart/react": "^1.1.1",
     "@radix-ui/colors": "^0.1.8",
+    "@radix-ui/react-icons": "^1.3.0",
```
Comment (kavinvenkatachalam) [inline frontend/package.json:12]: Can we remove this package?

File: frontend/package.json [hunk 2]
```diff
@@ -102,6 +106,8 @@
     "semver": "^7.3.8",
     "string-hash": "^1.1.3",
     "superstruct": "^1.0.3",
+    "tailwind-merge": "^2.2.1",
```
Comment (kavinvenkatachalam) [inline frontend/package.json:109]: If possible, we can avoid this and create our own util function

File: frontend/package.json [hunk 3]
```diff
@@ -23,7 +25,9 @@
     "array-move": "^4.0.0",
     "axios": "^1.3.3",
     "bootstrap": "^5.2.3",
+    "class-variance-authority": "^0.7.0",
     "classnames": "^2.3.2",
+    "clsx": "^2.1.0",
```
Comment (kavinvenkatachalam) [inline frontend/package.json:30]: Can we use `classnames` package which we already have?

File: frontend/src/App/App.jsx [hunk 4]
```diff
@@ -31,10 +31,8 @@ import 'react-tooltip/dist/react-tooltip.css';
 import { getWorkspaceIdOrSlugFromURL } from '@/_helpers/routes';
 import ErrorPage from '@/_components/ErrorComponents/ErrorPage';
 import WorkspaceConstants from '@/WorkspaceConstants';
-import { useAppDataStore } from '@/_stores/appDataStore';
```
Comment (kavinvenkatachalam) [inline frontend/src/App/App.jsx:34]: Can we revert this?

File: frontend/src/_styles/tabler.scss [hunk 5]
```diff
@@ -6,5 +6,19088 @@
...
...ound:rgba(255,255,255,.1)!important}.scrollable{overflow-x:hidden;overflow-y:auto;-webkit-overflow-scrolling:touch}.scrollable.hover{overflow-y:hidden}.scrollable.hover>*{margin-top:-1px}.scrollable.hover:active,.scrollable.hover:focus,.scrollable.hover:hover{overflow:visible;overflow-y:auto}.touch .scrollable{overflow-y:auto!important}.scroll-x,.scroll-y{overflow:hidden;-webkit-overflow-scrolling:touch}.scroll-y{overflow-y:auto}.scroll-x{overflow-x:auto}.no-scroll{overflow:hidden}.w-0{width:0!important}.h-0{height:0!important}.w-1{width:.25rem!important}.h-1{height:.25rem!important}.w-2{width:.5rem!important}.h-2{height:.5rem!important}.w-3{width:1rem!important}.h-3{height:1rem!important}.w-4{width:2rem!important}.h-4{height:2rem!important}.w-5{width:4rem!important}.h-5{height:4rem!important}.w-auto{width:auto!important}.h-auto{height:auto!important}.w-px{width:1px!important}.h-px{height:1px!important}.w-full{width:100%!important}.h-full{height:100%!important}.opacity-0{opacity:0!important}.opacity-5{opacity:.05!important}.opacity-10{opacity:.1!important}.opacity-15{opacity:.15!important}.opacity-20{opacity:.2!important}.opacity-25{opacity:.25!important}.opacity-30{opacity:.3!impo
```
Comment (kavinvenkatachalam) [inline frontend/src/_styles/tabler.scss:9]: Can we revert this one as well?

File: frontend/src/components/ui/Button/button.jsx [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+import PropTypes from 'prop-types';
+import { cn } from '@/lib/utils';
+import Loader from '../utilComponents/loader';
+import SolidIcon from '@/_ui/Icon/SolidIcons';
+// eslint-disable-next-line import/no-unresolved
+import { Slot } from '@radix-ui/react-slot';
+// eslint-disable-next-line import/no-unresolved
+import { cva } from 'class-variance-authority';
+import './button.scss';
+import { getDefaultIconFillColor, defaultButtonFillColour, getIconSize, getButtonWidth } from './buttonUtils.js';
```
Comment (kavinvenkatachalam) [inline frontend/src/components/ui/Button/button.jsx:11]:
```suggestion
import { getDefaultIconFillColor, defaultButtonFillColour, getIconSize } from './buttonUtils.js';
```

File: frontend/src/components/ui/Button/button.jsx [hunk 7]
```diff
@@ -0,0 +1,207 @@
+import React, { forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import { cn } from '@/lib/utils';
+import Loader from '../utilComponents/loader';
```
Comment (kavinvenkatachalam) [inline frontend/src/components/ui/Button/button.jsx:4]: We can create an alias and import it like `@/components/ui/utilComponents/loader`

File: frontend/src/components/ui/Button/button.jsx [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+      <Comp
+        className={cn(buttonVariants({ variant, size, iconOnly, className }))}
+        ref={ref}
+        disabled={disabled}
+        {...props}
+      >
+        {isLoading ? (
+          <div className="tw-flex tw-justify-center tw-items-center">
+            <Loader color={iconFillColor} width={getIconSize(size)} />
+            <a className="tw-invisible">{props.children}</a>
```
Comment (kavinvenkatachalam) [inline frontend/src/components/ui/Button/button.jsx:157]: Are we adding this to maintain the width?

File: frontend/src/components/ui/Button/buttonUtils.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        return 'tw-px-[20px]';
+      case 'default':
+        return 'tw-px-[12px]';
+      case 'medium':
+        return 'tw-px-[10px]';
+      case 'small':
+        return 'tw-px-[8px]';
+    }
+  }
+};
```
Comment (kavinvenkatachalam) [inline frontend/src/components/ui/Button/buttonUtils.js:59]: Can we remove this if it is not ued anywhere?

File: frontend/src/components/ui/input.jsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        className
+      )}
+      ref={ref}
+      {...props}
+    />
+  );
+});
+Input.displayName = 'Input';
+
+export { Input };
```
Comment (kavinvenkatachalam) [inline frontend/src/components/ui/input.jsx:20]: We can remove this if it is not needed

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@stepinfwd Suggested some changes. Please take a look. Can we `Capitalize` the component name?

File: frontend/src/components/ui/Button/button.jsx (hunk 8 above)
Comment (stepinfwd) [reply frontend/src/components/ui/Button/button.jsx:157]: Yes , for getting icon height and width

File: frontend/package.json (hunk 2 above)
Comment (stepinfwd) [reply frontend/package.json:109]: Will be keeping this

File: frontend/src/_styles/tabler.scss (hunk 5 above)
Comment (stepinfwd) [reply frontend/src/_styles/tabler.scss:9]: Error is being thrown from `autoprefixer` without this change

**Review by kavinvenkatachalam — DISMISSED** [review]

**Review by gsmithun4 — APPROVED** [review]

**Pull request merged**